lib/presentation/widgets/recent_search.dart

```diff
@@ -1,14 +1,91 @@
-import 'package:flutter/cupertino.dart';
+import 'package:bus_proj/bloc/bus_bloc.dart';
+import 'package:bus_proj/bloc/bus_state.dart';
+import 'package:bus_proj/constants/app_color.dart';
+import 'package:bus_proj/presentation/presentation.dart';
+import 'package:bus_proj/presentation/widgets/common/app_custom_route.dart';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:phosphor_flutter/phosphor_flutter.dart';
 
 class RecentSearchWidget extends StatelessWidget {
   const RecentSearchWidget({super.key});
 
   @override
   Widget build(BuildContext context) {
-    return const Column(
-      children: [
-        Text('Recent Search'),
-      ],
+    final busBloc = BlocProvider.of<BusBloc>(context)..getRecentSearch();
+    final double screenWidth = MediaQuery.of(context).size.width;
+    final double screenHeight = MediaQuery.of(context).size.height;
+
+    return BlocConsumer<BusBloc, BusState>(
+      listener: (context, state) {},
+      builder: (context, state) {
+        return SingleChildScrollView(
+          child: Container(
+            width: screenWidth,
+            decoration: BoxDecoration(
+              color: Colors.white,
+              borderRadius: BorderRadius.circular(20),
+            ),
+            child: Padding(
+              padding: const EdgeInsets.all(15.0),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      const Text(
+                        'Search History',
+                        style: TextStyle(
+                          fontWeight: FontWeight.bold,
+                          fontSize: 16,
+                        ),
+                      ),
+                      TextButton(
+                        onPressed: () {
+                          busBloc.clearRecentSearch();
+                        },
+                        child: const Text(
+                          'Clear All',
+                          style: TextStyle(
+                            color: Colors.red,
+                          ),
+                        ),
+                      ),
+                    ],
+                  ),
+                  SizedBox(
+                    height: screenHeight,
```

Comment:
height should not be equal to screenHeight but the total height of  the children its having

Reply:
Done